submissions/kirill8210/memory-game/index.js

```diff
@@ -0,0 +1,81 @@
+import arrayNumbers from './json.js';
+
+const mainField = document.querySelector('.main');
+const doubleNumbers = [...arrayNumbers, ...arrayNumbers];
+
+const createCard = (arr) => {
+    const {value, id} = arr;
+    const cards = document.createElement('div');
+    cards.className = 'card';
+    cards.dataset.dataid = `${id}`;
+
+    cards.insertAdjacentHTML('afterbegin', `
+        <div id="${id}" class="flip-container">
+            <div class="flipper">
+                <div class="front">
+                    <div class="front_card">You number</div>
+                </div>
+                <div class="back">
+                    <div class="back_card">${value}</div>
+                </div>
+            </div>
+        </div>
+    `);
+
+    return cards
+};
+
+const shuffleCards = (arr) => {
+    const card = arr.sort(() => 0.5 - Math.random()).map(createCard);
+    mainField.append(...card);
+};
+
+shuffleCards(doubleNumbers);
+
+const toggleCards = document.querySelectorAll('.flipper');
+let countFlipCards = 0;
+let openedCardId = 0;
```

Comment:
Look, you have `openedCardId` that stores an ID of an opened card and then you have `oneCard` and `twoCard` variables that store cards themselves. There are 3 variables and it could be changed to 1 variable.

Let's see what functionality you suppose them to have:
1. They should indicate if there is a card open right now or not.
2. They should store what Id exactly an opened card has.
3. If the user opens a second card they should give you an option to compare IDs of an opened card and the newly clicked one.

Now, onto making it more clear.

We have 2 options:
First, we can get rid of the `openedCardId` variable, since from what I see in your code, it can only store a real ID when `oneCard` is not empty. But the problem is, `oneCard` stores the whole card. With ID inside it. So you're just doubling data, no need for that.

Second (and I think this one is better), we can get rid of `openedCardId` and `twoCard` completely. The `twoCard` variable in global scope doesn't make any sence, it is only used inside the `flipCard` function. Which means you can only store information about a single opened card in the global scope. And the opened card ID is stored in the `oneCard` anyway.

So you can just leave a single variable - `openedCard` and make it `null` by default.

When user clicks a card you do `if (openedCard)...`, and if it's empty, open the card and store it in this variable.
When user clicks another card, you just operate with a second card in bounds of the `flipCard` function, no need to store it somewhere. Check if the card is the same one and do nothing it that's the case. If not, check if they're equal. If they are - make them dissapear and place `null` to `openedCard` once again.